### PR TITLE
Make CropTools Gobject-style constructed

### PR DESCRIPTION
**Pull request by ZenitsuDev**
Next to EditingTools, make children classes gobject style constructed. Starting with CropTools. Also fixed a simple block in AdjustTools.

### CODE REVIEW

File: src/editing_tools/CropTool.vala [hunk 1]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
-            bottom = y;
-            if (get_constraint_aspect_ratio () != ANY_ASPECT_RATIO) {
-                float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
-                right = left + ((int) new_width);
-            }
-            break;
+            case BoxLocation.LEFT_SIDE:
+                left = x;
+                if (get_constraint_aspect_ratio () != ANY_ASPECT_RATIO) {
+                    float new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:921]: Can use var here:
```suggestion
                    var new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```

Other examples exist but not commented here as not changed

File: src/editing_tools/CropTool.vala [hunk 2]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
 
-        case BoxLocation.BOTTOM_LEFT:
-            if (get_constraint_aspect_ratio () == ANY_ASPECT_RATIO) {
+            case BoxLocation.BOTTOM_SIDE:
                 bottom = y;
-                left = x;
-            } else {
-                if (y < eval_radial_line (center_x, center_y, left, bottom, x)) {
+                if (get_constraint_aspect_ratio () != ANY_ASPECT_RATIO) {
+                    float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:945]:
```suggestion
                    var new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 3]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
+            case BoxLocation.TOP_LEFT:
+                if (get_constraint_aspect_ratio () == ANY_ASPECT_RATIO) {
+                    top = y;
                     left = x;
-                    float new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
-                    bottom = top + ((int) new_height);
                 } else {
+                    if (y < eval_radial_line (center_x, center_y, left, top, x)) {
+                        top = y;
+                        float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:957]:
```suggestion
                        var new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 4]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
+                        left = right - ((int) new_width);
+                    } else {
+                        left = x;
+                        float new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:961]:
```suggestion
                        var new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 5]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
+            case BoxLocation.BOTTOM_LEFT:
+                if (get_constraint_aspect_ratio () == ANY_ASPECT_RATIO) {
                     bottom = y;
-                    float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
-                    left = right - ((int) new_width);
+                    left = x;
+                } else {
+                    if (y < eval_radial_line (center_x, center_y, left, bottom, x)) {
+                        left = x;
+                        float new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:974]:
```suggestion
                        var new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 6]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
+                        bottom = top + ((int) new_height);
+                    } else {
+                        bottom = y;
+                        float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:978]:
```suggestion
                        var new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 7]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
-                    float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
-                    right = left + ((int) new_width);
-                } else {
                     right = x;
-                    float new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
-                    top = bottom - ((int) new_height);
+                } else {
+                    if (y < eval_radial_line (center_x, center_y, right, top, x)) {
+                        top = y;
+                        float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:991]:
```suggestion
                        var new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 8]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
+                        right = left + ((int) new_width);
+                    } else {
+                        right = x;
+                        float new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:995]:
```suggestion
                        var new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 9]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
                     right = x;
-                    float new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
-                    bottom = top + ((int) new_height);
                 } else {
-                    bottom = y;
-                    float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
-                    right = left + ((int) new_width);
+                    if (y < eval_radial_line (center_x, center_y, right, bottom, x)) {
+                        right = x;
+                        float new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:1008]:
```suggestion
                        var new_height = ((float) (right - left)) / get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 10]
```diff
@@ -889,159 +915,165 @@ public class EditingTools.CropTool : EditingTool {
...
+                        bottom = top + ((int) new_height);
+                    } else {
+                        bottom = y;
+                        float new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:1012]:
```suggestion
                        var new_width = ((float) (bottom - top)) * get_constraint_aspect_ratio ();
```

File: src/editing_tools/CropTool.vala [hunk 11]
```diff
@@ -1064,41 +1096,45 @@ public class EditingTools.CropTool : EditingTool {
...
+
+                case BoxLocation.RIGHT_SIDE:
+                case BoxLocation.TOP_RIGHT:
+                case BoxLocation.BOTTOM_RIGHT:
+                    if (width < CROP_MIN_SIZE) {
+                        right = left + CROP_MIN_SIZE;
+                    }
+                    break;
+
+                default:
```
Comment (jeremypw) [inline src/editing_tools/CropTool.vala:1115]: Elementary code style: blank line after `if (){}`
```suggestion
                    if (width < CROP_MIN_SIZE) {
                        right = left + CROP_MIN_SIZE;
                    }

                    break;
                default:
```
Many examples not additionally commented

**Review by jeremypw — CHANGES_REQUESTED** [review]
Just some nitpicking comments on codestyle, otherwise no regressions found.

File: src/editing_tools/CropTool.vala (hunk 1 above)
Comment (ZenitsuDev) [reply src/editing_tools/CropTool.vala:921]: Okay. I will update it then😁

**Comment by ZenitsuDev** [comment]
Suggestions committed! Thanks @jeremypw

**Review by jeremypw — APPROVED** [review]
You didn't insert blank lines after every if clause but as that is not the main purpose of this PR I think it can be left for another PR.  Thanks!

**Pull request merged**